sql/core/src/main/scala/org/apache/spark/sql/execution/CollectMetricsExec.scala

```diff
@@ -33,8 +35,65 @@ case class CollectMetricsExec(
     child: SparkPlan)
   extends UnaryExecNode {
 
+  /**
+   * The specific implementation of AggregatingAccumulator for calling `merge` on executor side.
+   */
+  private class MetricsAggregatingAccumulator(
```

Comment:
Does it worth the complexity? Where do we use `AggregatingAccumulator` except for `CollectMetricsExec`?

Reply:
This is to response to the [comment](https://github.com/apache/spark/pull/32786#discussion_r647338534) from @hvanhovell .

Reply:
If you prefer a more elegant solution, I think improving the accumulator framework to introduce "no update via heartbeat" is a better approach.